tests/unit/lms/services/grouping_test.py

```diff
@@ -14,7 +15,7 @@ class TestGroupingService:
     TOOL_CONSUMER_INSTANCE_GUID = "t_c_i_guid"
 
     def test_upsert_with_parents_inserts(self, svc, db_session):
-        course = factories.Course()
+        course = factories.Course(application_instance=svc.application_instance)
```

Comment:
This test and a couple of others below were creating groupings with different `application_instance`'s than their parents and so now crashing, so I fixed them.

The way it works is:

1. If you don't pass any `application_instance` to `factories.Course()` it'll generate a new one, and so `course` (which is going to be passed as the `parent` to `GroupingService.upsert_with_parent()` below) would end up with a different `application_instance` than the one that `GroupingService` uses for the new child grouping that it creates

2. `GroupingService.upsert_with_parent()` uses the current `ApplicationInstance` (from `ApplicationInstanceService.get_current()` as the `Grouping.application_instance` for any new child groupings that it creates.

3. In the tests the real `GroupingService` uses the mock `ApplicationInstanceService` from the `application_instance_service` fixture. So it uses `application_instance_service.get_current.return_value` as the `application_instance` for any new child groupings it creates

4. The mock `application_instance_service.get_current.return_value` also happens to be the same object as the `application_instance` fixture

Rather than having the tests implicitly rely on these connections between the `GroupingService` under test and the mock `application_instance_service` and `application_instance` fixtures, I made `GroupingService.application_instance` public so the tests can access `svc.application_instance`.

Reply:
I think this good. Explicit is better than implicit. I mentioned below that this is something we now do three times, so I think a fixture would be in order.

It might be good to add a comment like `# Ensure the course is attached to the same application instance the service uses` or something.

Reply:
Mentioned elsewhere but these tests later got refactored (introducing a caller pattern) in a way that removed this duplication so that a fixture was no longer needed